test/config/config.ts

```diff
@@ -1,98 +1,346 @@
-    port: process.env.DB2_PORT_50000_TCP_PORT || process.env.SEQ_DB2_PORT || process.env.SEQ_PORT
-      || process.env.IBM_DB_PORT || 50_000,
-    pool: {
-      max: Number(process.env.SEQ_DB2_POOL_MAX || process.env.SEQ_POOL_MAX || 5),
-      idle: Number(process.env.SEQ_DB2_POOL_IDLE || process.env.SEQ_POOL_IDLE || 3000),
+    // DB2 v11.5.5 and older: 50000; v11.5.6 and newer: 25000
+    native: { port: 50_000 },
+    default: {
+      database: env.SEQ_DB2_DB || env.SEQ_DB || env.IBM_DB_DBNAME || 'testdb',
+      username: env.SEQ_DB2_USER || env.SEQ_USER || env.IBM_DB_UID || 'db2inst1',
+      password: env.SEQ_DB2_PW || env.SEQ_PW || env.IBM_DB_PWD || 'password',
+      host: env.DB2_PORT_50000_TCP_ADDR || env.SEQ_DB2_HOST || env.SEQ_HOST || env.IBM_DB_HOSTNAME || '127.0.0.1',
+      port: env.DB2_PORT_50000_TCP_PORT || env.SEQ_DB2_PORT || env.SEQ_PORT || env.IBM_DB_PORT, // || 50_000,
+      pool: {
+        max: Number(env.SEQ_DB2_POOL_MAX || env.SEQ_POOL_MAX || 5),
+        idle: Number(env.SEQ_DB2_POOL_IDLE || env.SEQ_POOL_IDLE || 3000),
+      },
+    },
+    11.5: {
+      port: env.DB2_PORT_50000_TCP_PORT || env.SEQ_DB2_PORT || env.SEQ_PORT || env.IBM_DB_PORT || 50_000,
     },
   },
+
+  //  ╦╔╗ ╔╦╗╦  ┌─┐┌─┐┌┬┐┌┬┐┬┌┐┌┌─┐┌─┐
+  //  ║╠╩╗║║║║  └─┐├┤  │  │ │││││ ┬└─┐
+  //  ╩╚═╝╩ ╩╩  └─┘└─┘ ┴  ┴ ┴┘└┘└─┘└─┘
+
   ibmi: {
-    database: env.SEQ_IBMI_DB || env.SEQ_DB,
-    username: process.env.SEQ_IBMI_USER || process.env.SEQ_USER,
-    password: process.env.SEQ_IBMI_PW || process.env.SEQ_PW,
-    pool: {
-      max: Number(env.SEQ_IBMI_POOL_MAX || env.SEQ_POOL_MAX || env.SEQ_POOL_MAX || 5),
-      idle: Number(env.SEQ_IBMI_POOL_IDLE || env.SEQ_POOL_IDLE || 3000),
-    },
-    dialectOptions: {
-      odbcConnectionString: env.SEQ_IBMI_CONN_STR,
+    native: { port: 8471 },
+    default: {
+      database: env.SEQ_IBMI_DB || env.SEQ_DB,
+      username: env.SEQ_IBMI_USER || env.SEQ_USER,
+      password: env.SEQ_IBMI_PW || env.SEQ_PW,
+      pool: {
+        max: Number(env.SEQ_IBMI_POOL_MAX || env.SEQ_POOL_MAX || env.SEQ_POOL_MAX || 5),
+        idle: Number(env.SEQ_IBMI_POOL_IDLE || env.SEQ_POOL_IDLE || 3000),
+      },
+      dialectOptions: {
+        odbcConnectionString: env.SEQ_IBMI_CONN_STR,
+      },
     },
   },
 };
+
+//   ██████╗ ██████╗ ███╗   ██╗███████╗██╗ ██████╗     ███████╗██╗   ██╗███╗   ██╗ ██████╗
+//  ██╔════╝██╔═══██╗████╗  ██║██╔════╝██║██╔════╝     ██╔════╝██║   ██║████╗  ██║██╔════╝
+//  ██║     ██║   ██║██╔██╗ ██║█████╗  ██║██║  ███╗    █████╗  ██║   ██║██╔██╗ ██║██║
+//  ██║     ██║   ██║██║╚██╗██║██╔══╝  ██║██║   ██║    ██╔══╝  ██║   ██║██║╚██╗██║██║
+//  ╚██████╗╚██████╔╝██║ ╚████║██║     ██║╚██████╔╝    ██║     ╚██████╔╝██║ ╚████║╚██████╗
+//   ╚═════╝ ╚═════╝ ╚═╝  ╚═══╝╚═╝     ╚═╝ ╚═════╝     ╚═╝      ╚═════╝ ╚═╝  ╚═══╝ ╚═════╝
+//
+
+export function Config(dialect: string | undefined): Options {
```

Comment:
if `config` is a function it should not be Capitalized, it should also have a more explicit name such as `getConfig`, `buildConfig`